### PR TITLE
Update shuffle to use arc4random_uniform

### PR DESCRIPTION
**Pull request by kvnify**
Fixes issue #48.

### CODE REVIEW

File: Underscore/USArrayWrapper.m [hunk 1]
```diff
@@ -168,8 +168,9 @@ - (USArrayWrapper *)shuffle
 {
     NSMutableArray *result = [self.array mutableCopy];
 
+    NSInteger max = result.count;
```
Comment (robb) [inline Underscore/USArrayWrapper.m]: Can we make this variable `u_int32_t` and move the cast here? Also, can you `NSAssert` that count must be less than or equal to `UINT32_MAX`? Unlikely I know, but [stranger things have happened](http://googleresearch.blogspot.de/2006/06/extra-extra-read-all-about-it-nearly.html).

**Comment by robb** [comment]
Looks good, just one note

File: Underscore/USArrayWrapper.m (hunk 1 above)
Comment (kvnify) [inline Underscore/USArrayWrapper.m]: No problem. Although my reading suggests that the assert might still be overkill.  I don't thin we'll have iOS/MacOS hardware that will support that many elements stored in an array anytime soon.

Comment (robb) [inline Underscore/USArrayWrapper.m]: Yeah, totally.

Comment (kvnify) [inline Underscore/USArrayWrapper.m]: If you're good with this I'm gonna merge it...and then #69 shortly thereafter.

**Comment by robb** [comment]
:sparkles:

**Pull request merged**